CMakeToolchain.txt

```diff
@@ -0,0 +1,3 @@
+set (CMAKE_SYSTEM_NAME Linux)
+
+set (CMAKE_SYSTEM_PROCESSOR arm)
```

Comment:
Does this mean Aarch64 is not supported?

Reply:
Given that it works with Aarch64 I'm not sure if this is needed or not. We have the same stuff in optee_client

Reply:
Let's remove it if it's unused, because it is misleading.

Reply:
OK

Reply:
That is used as a marker for later on make decisions in CMake to set architecture dependent flags etc, so I think it is safe remove it, since we don't use it. Igor please, chime in if I am wrong here.

The reason it works compile wise is because of the exports, you export CC, 
```bash
export CC="$HOME/devel/optee_projects/reference/toolchains/aarch32/bin/arm-linux-gnueabihf-gcc"
```

which in turns leads to this after running `cmake ..`:
```bash
CMAKE_C_COMPILER                 /home/jbech/devel/optee_projects/reference/toolchains/aarch32/bin/arm-linux-gnueabihf-gcc
```
So for aarch64, you instead export `aarch64-linux-gnu-gcc`.

Reply:
Using `CC` this way is actually not correct (what if we need to compile and run some tool on the build host?).

Reply:
It it interfere, then you can set it directly when running cmake, i.e., 
```bash
$ mkdir build && cd build
$ cmake -DCMAKE_C_COMPILER=/home/jbech/devel/optee_projects/reference/toolchains/aarch32/bin/arm-linux-gnueabihf-gcc ..
```